### PR TITLE
feat!: update to parser API v3

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR fixes that parser versions started to drift, as of `3.0.0-next-major-spec.9` we start using parser API v3 and no longer v2. This PR updates the template to use the newest parser API v3.

Related to https://github.com/asyncapi/html-template/issues/456
Supersedes https://github.com/asyncapi/html-template/pull/461

### CODE REVIEW

File: helpers/all.js
Comment (jonaslagoni) [inline helpers/all.js:137]: This is a big cause of problems i.e. https://github.com/asyncapi/html-template/issues/456 (have had it multiple times). So thought it was better to throw an error instead of generating empty HTML pages.

File: test/spec/asyncapi_v3.yml
Comment (jonaslagoni) [inline test/spec/asyncapi_v3.yml:3]: Changed the test document to the v2 document, just converted to v3.

**Review by jonaslagoni — COMMENTED** [review]
The test will fail until the following PRs have been merged:
- https://github.com/smoya/multi-parser-js/pull/5
- https://github.com/asyncapi/generator/pull/1066

File: helpers/all.js
Comment (derberg) [inline helpers/all.js:137]:
```suggestion
  if(stringifiedDoc === undefined) throw new Error("Unable to stringify parsed AsyncAPI document passed by the generator. Please report an issue in https://github.com/asyncapi/html-template repository.")
```

**Review by derberg — DISMISSED** [review]
I commit directly an update to error message

**Review by fmvilas — APPROVED** [review]

**Comment by jonaslagoni** [comment]
@derberg @fmvilas so the test actually blocks the release right now.

Do you want to make the CI check optional for a bit or?

**Comment by jonaslagoni** [comment]
Ready to be reviewed for a final time @fmvilas @derberg

**Review by fmvilas — APPROVED** [review]

**Pull request merged**
